src/components/atom/button/index.tsx

```diff
@@ -0,0 +1,22 @@
+import { ReactComponent as AddIcon } from "../../../assets/img/addIcon.svg";
+import "./styles.scss";
+
+type ButtonProps = {
+  text: string;
+  hasAddIcon?: boolean;
+  onClick: () => void;
+  type: HTMLButtonElement["type"];
+  disabled?: boolean;
+  styles: string;
+};
+
+const Button = ({ text, hasAddIcon, onClick, type, disabled, styles, ...rest }: ButtonProps) => {
+  return (
+    <button className={styles} onClick={onClick} disabled={disabled} type={type} {...rest}>
```

Comment:
One suggestion for future cases. When u have to disable the component u have to prevent click or some other actions. Right now if I go to the inspect element in the browser and remove disable property, I can easily click on btn and trigger the action. In order to prevent that u have to make smth like this 
onClick={() => !disabled ? onClick() : null} With this u will prevent the prev case cuz on the mount component we will check if we are gonna append some action to the button or we are not. Even though u removed disabled from the inspect element, the component is already mounted without the "click" action